### PR TITLE
fix(cli): stranded classification follow-up — deduped lookups, outage wording, unit tests

### PR DESCRIPTION
**Pull request by irfndi**
Follow-up on the kilo review comments left on #188 (which merged before the comments landed). Addresses all three:

1. **Per-mint fallback duplicated full price-chain fetches per settlement** (comment on cli/status.ts:373): price and decimals lookups are now resolved ONCE per unique mint into maps; every settlement classifies from those maps. The batch-defect fallback fires one fetch per unique mint (deduplicated), not per settlement.
2. **Unavailable line blamed the price source** (cli/status.ts:404): a decimals/RPC outage was misattributed. Wording is now outage-neutral: `price/decimals lookup unreachable`.
3. **The three-channel split had no test coverage** (cli/status.ts:356): the classification is extracted into a pure, exported `classifyStrandedSettlement` helper with a dedicated unit suite — stranded/dust boundary (incl. exact-cutoff), price-typed-failure vs decimals-typed-failure vs defect vs no-price, NaN amounts, and outage-over-unpriceable precedence.

Verification: lint clean, full suite 121 files / 1571 tests.

## Summary by Sourcery

Improve stranded settlement classification in the CLI, deduplicating per-mint lookups and tightening outage handling and reporting.

New Features:
- Expose a pure classifyStrandedSettlement helper to encapsulate stranded vs dust vs unavailable vs unpriceable classification logic.

Bug Fixes:
- Prevent duplicate per-mint price and decimals fetches by sharing lookup results across stranded settlements.
- Correct stranded settlement outage messaging to attribute failures generically to price/decimals lookups rather than the price source alone.

Enhancements:
- Introduce a three-channel stranded classification model with explicit states for ok, unavailable, and unpriceable lookups.

Tests:
- Add unit tests for classifyStrandedSettlement covering dust cutoff boundaries, typed failures, unpriceable tokens, NaN amounts, and outage precedence.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved settlement status accuracy for balances near dust thresholds.
  * Clearly distinguishes unavailable price or decimals data from values that cannot be priced.
  * Handles malformed, non-finite, and adapter error responses more reliably.
  * Improved failure messages to identify whether price or decimals information is unavailable.
* **Performance**
  * Reduced redundant lookups and improved settlement valuation efficiency.

### CODE REVIEW

File: cli/status.ts [hunk 1]
```diff
@@ -350,76 +396,98 @@ network; with no stranded settlements it is fully offline.`,
...
+              const entries = candidateMints.map((mint) => {
+                const price = prices[mint] ?? 0;
+                return [
+                  mint,
+                  { state: (price > 0 ? "ok" : "unpriceable") as StrandedLookupState, value: price },
+                ] as const;
+              });
+              return new Map(entries);
+            }),
+            Effect.catch(() =>
```
[the remark this answers is not in the document]
Comment (irfndi) [reply cli/status.ts]: Fixed: the dead price-typed-failure branch is removed — fetchTokenPrices never fails (all sources catch internally, unresolved = 0), so a price-provider outage is indistinguishable from a genuinely unquotable mint at the adapter API and classifies as Unpriceable, which is factually true at query time. The code and comments now say exactly that; the Unavailable channel is driven solely by the decimals/RPC lookup, which does fail typed.

File: cli/status.ts [hunk 2]
```diff
@@ -350,76 +396,98 @@ network; with no stranded settlements it is fully offline.`,
...
+          yield* Effect.all(
+            candidateMints
+              .filter((mint) => priceLookup.get(mint)?.state === "ok")
+              .map((mint) =>
+                adapter.getTokenDecimals(mint).pipe(
+                  Effect.map((decimals) => {
+                    const state: StrandedLookupState = decimals > 0 ? "ok" : "unpriceable";
+                    return { mint, state, value: decimals };
+                  }),
+                  Effect.catch(() =>
```
[the remark this answers is not in the document]
Comment (irfndi) [reply cli/status.ts]: Fixed: decimalsFailureState now matches the adapter's specific "Cannot resolve decimals for mint X via Helius or standard RPC" message → Unpriceable (permanent, a retry can never succeed), while other typed errors (RPC outage/timeout) → Unavailable. Unit tests added for the mapping (unresolvable vs outage vs non-Error).

File: cli/status.ts [hunk 3]
```diff
@@ -336,90 +396,112 @@ network; with no stranded settlements it is fully offline.`,
...
-                const amountNum = Number(settlement.amountAtomic);
-                if (!Number.isFinite(amountNum)) {
-                  return { settlement, kind: "unpriceable" as const };
-                }
-                const valueUsd = (amountNum / 10 ** decimalsResult.decimals) * price;
+          // Price channel: fetchTokenPrices NEVER fails — every source
+          // (Helius/Jupiter/CoinGecko) catches its own errors and returns {},
+          // and unresolved mints come back as price 0 — so a total
+          // price-provider outage is INDISTINGUISHABLE from a genuinely
+          // unquotable mint at this API and classifies as Unpriceable (the
```
[the remark this answers is not in the document]
Comment (irfndi) [reply cli/status.ts:411]: Fixed in follow-up PR #193: both price calls (batched + per-mint fallback) now pass { useFallback: false } — the fabricated fallback prices (SOL at $165 etc.) can no longer classify stranded capital at made-up values during an outage; it resolves to price 0 → Unpriceable, matching the wallet-reconciliation path.
